File: Csharp/fr.eulbobo.dojo.byron/fr.eulbobo.dojo.byron.tests/ByronRecreationTest.cs
using fr.eulbobo.dojo.byron.domain;
using Moq;

namespace fr.eulbobo.dojo.byron.tests
{
    public class ByronRecreationTest
    {
        /**
         * 1    | Is he aware the walls are exceptionnaly thin ?           | +1 Stress                 |
         */
        [Test]
        public void shouldAdd1StressWhenRolls1()
        {
            var sanity = new Mock<Sanity>();
            var eventLog = new Mock<EventLog>();
            ByronEvent @event =new ByronRecreation(eventLog.Object);
            @event.ApplyDiceTo(DiceRoll.Of(1), sanity.Object);

            eventLog.Verify(evtl => evtl.Log("Is he aware the walls are exceptionnaly thin ?"));
            sanity.Verify(s => s.IncreaseStress(), Times.Once);
            sanity.VerifyNoOtherCalls();
        }

        /**
         * 2    | He's made a mess of your desk in the process             | -1 MasterPiece            |
         */
        [Test]
        public void shouldReduce1MasterPieceWhenRolls2()
        {
            var sanity = new Mock<Sanity>();
            var eventLog = new Mock<EventLog>();
            ByronEvent @event =new ByronRecreation(eventLog.Object);
            @event.ApplyDiceTo(DiceRoll.Of(2), sanity.Object);

            eventLog.Verify(evtl => evtl.Log("He's made a mess of your desk in the process"));
            sanity.Verify(s => s.DecreaseMasterpiece(), Times.Once);
            sanity.VerifyNoOtherCalls();
        }

        /**
         * 3    | May he borrow your husband? Of course.                   | -1 Stress <br> +1 Scandal |
         */
        [Test]
        public void shouldReduceStressByOneAndIncreaseScandalByOneWhenRolls3()
        {
            var sanity = new Mock<Sanity>();
            var eventLog = new Mock<EventLog>();
            ByronEvent @event =new ByronRecreation(eventLog.Object);
            @event.ApplyDiceTo(DiceRoll.Of(3), sanity.Object);

            eventLog.Verify(evtl => evtl.Log("May he borrow your husband? Of course."));
            sanity.Verify(s => s.DecreaseStress(), Times.Once);
            sanity.Verify(s => s.IncreaseScandal(), Times.Once);
            sanity.VerifyNoOtherCalls();
        }

        /**
         * 4    | His Half-sister is here, and they are *FAR* too intimate | +2 Scandal                |
         */
        [Test]
        public void shouldAdd2ScandalWhenRolls4()
        {
            var sanity = new Mock<Sanity>();
            var eventLog = new Mock<EventLog>();
            ByronEvent @event =new ByronRecreation(eventLog.Object);
            @event.ApplyDiceTo(DiceRoll.Of(4), sanity.Object);

            eventLog.Verify(evtl => evtl.Log("His Half-sister is here, and they are FAR too intimate"));
            sanity.Verify(s => s.IncreaseScandal(), Times.Exactly(2));
            sanity.VerifyNoOtherCalls();
        }

        /**
         * 5    | You are weary of listening to the tales of his exploits | +1 Stress                 |
         */
        [Test]
        public void shouldAdd1StressWhenRolls5()
        {
            var sanity = new Mock<Sanity>();
            var eventLog = new Mock<EventLog>();
            ByronEvent @event =new ByronRecreation(eventLog.Object);
            @event.ApplyDiceTo(DiceRoll.Of(5), sanity.Object);

            eventLog.Verify(evtl => evtl.Log("You are weary of listening to the tales of his exploits"));
            sanity.Verify(s => s.IncreaseStress(), Times.Once);
            sanity.VerifyNoOtherCalls();
        }

        /**
         * 6    | He Makes and excellent muse on occasion                  | -1 Stress <br> +2 Masterpiece |
         */
        [Test]
        public void shouldDecrease1StressAndIncrease2MasterpieceWhenRolls6()
        {
            var sanity = new Mock<Sanity>();
            var eventLog = new Mock<EventLog>();
            ByronEvent @event = new ByronRecreation(eventLog.Object);
            @event.ApplyDiceTo(DiceRoll.Of(6), sanity.Object);

            eventLog.Verify(evtl => evtl.Log("He makes and excellent muse on occasion"));            
            sanity.Verify(s => s.DecreaseStress(), Times.Once);
            sanity.Verify(s => s.IncreaseMasterpiece(), Times.Exactly(2));
            sanity.VerifyNoOtherCalls();
        }
    }
}

File: java/src/test/java/fr/eulbobo/dojo/byron/domain/ByronRecreationTest.java
package fr.eulbobo.dojo.byron.domain;

import org.junit.jupiter.api.Test;
import org.mockito.Mockito;

import static org.mockito.ArgumentMatchers.eq;
import static org.mockito.Mockito.*;


class ByronRecreationTest {

    /**
     * 1    | Is he aware the walls are exceptionnaly thin ?           | +1 Stress                 |
     */
    @Test
    void shouldAdd1StressWhenRolls1() {
        Sanity sanity = Mockito.mock(Sanity.class);
        EventLog eventLog = Mockito.mock(EventLog.class);
        ByronEvent event = new ByronRecreation(eventLog);
        event.applyDiceTo(DiceRoll.of(1), sanity);

        verify(eventLog).log(eq("Is he aware the walls are exceptionnaly thin ?"));
        verify(sanity, times(1)).increaseStress();
        verifyNoMoreInteractions(sanity);
    }

    /**
     * 2    | He's made a mess of your desk in the process             | -1 MasterPiece            |
     */
    @Test
    void shouldReduce1MasterPieceWhenRolls2() {
        Sanity sanity = Mockito.mock(Sanity.class);
        EventLog eventLog = Mockito.mock(EventLog.class);
        ByronEvent event = new ByronRecreation(eventLog);
        event.applyDiceTo(DiceRoll.of(2), sanity);

        verify(eventLog).log(eq("He's made a mess of your desk in the process"));
        verify(sanity, times(1)).decreaseMasterpiece();
        verifyNoMoreInteractions(sanity);
    }

    /**
     * 3    | May he borrow your husband? Of course.                   | -1 Stress <br> +1 Scandal |
     */
    @Test
    void shouldReduceStressByOneAndIncreaseScandalByOneWhenRolls3() {
        Sanity sanity = Mockito.mock(Sanity.class);
        EventLog eventLog = Mockito.mock(EventLog.class);
        ByronEvent event = new ByronRecreation(eventLog);
        event.applyDiceTo(DiceRoll.of(3), sanity);

        verify(eventLog).log(eq("May he borrow your husband? Of course."));
        verify(sanity, times(1)).decreaseStress();
        verify(sanity, times(1)).increaseScandal();
        verifyNoMoreInteractions(sanity);
    }

    /**
     * 4    | His Half-sister is here, and they are *FAR* too intimate | +2 Scandal                |
     */
    @Test
    void shouldAdd2ScandalWhenRolls4() {
        Sanity sanity = Mockito.mock(Sanity.class);
        EventLog eventLog = Mockito.mock(EventLog.class);
        ByronEvent event = new ByronRecreation(eventLog);
        event.applyDiceTo(DiceRoll.of(4), sanity);

        verify(eventLog).log(eq("His Half-sister is here, and they are FAR too intimate"));
        verify(sanity, times(2)).increaseScandal();
        verifyNoMoreInteractions(sanity);
    }

    /**
     * 5    | You are weary of listening to the tales of his exploits | +1 Stress                 |
     */
    @Test
    void shouldAdd1StressWhenRolls5() {
        Sanity sanity = Mockito.mock(Sanity.class);
        EventLog eventLog = Mockito.mock(EventLog.class);
        ByronEvent event = new ByronRecreation(eventLog);
        event.applyDiceTo(DiceRoll.of(5), sanity);

        verify(eventLog).log(eq("You are weary of listening to the tales of his exploits"));
        verify(sanity, times(1)).increaseStress();
        verifyNoMoreInteractions(sanity);
    }

    /**
     * 6    | He Makes and excellent muse on occasion                  | -1 Stress <br> +2 Masterpiece |
     */
    @Test
    void shouldDecrease1StressAndIncrease2MasterpieceWhenRolls6() {
        Sanity sanity = Mockito.mock(Sanity.class);
        EventLog eventLog = Mockito.mock(EventLog.class);
        ByronEvent event = new ByronRecreation(eventLog);
        event.applyDiceTo(DiceRoll.of(6), sanity);

        verify(eventLog).log(eq("He makes and excellent muse on occasion"));
        verify(sanity, times(1)).decreaseStress();
        verify(sanity, times(2)).increaseMasterpiece();
        verifyNoMoreInteractions(sanity);
    }

}